src/NUnitEngine/nunit.engine.tests/Internal/FileSystemAccess/Default/FileTests.cs

```diff
@@ -0,0 +1,115 @@
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+namespace NUnit.Engine.Tests.Internal.FileSystemAccess.Default
+{
+    using NUnit.Engine.Internal.FileSystemAccess.Default;
+    using NUnit.Framework;
+    using System.Linq;
+    using System.Reflection;
+    using SIO = System.IO;
+
+    [TestFixture]
+    public sealed class FileTests
+    {
+        private string GetTestFileLocation()
+        {
+#if NETCOREAPP1_1
+            return Assembly.GetEntryAssembly().Location;
+#else
+            return Assembly.GetAssembly(typeof(FileTests)).Location;
+#endif
+        }
+
+        [Test]
+        public void Init()
+        {
+            var path = this.GetTestFileLocation();
+            var parent = SIO.Path.GetDirectoryName(path);
+
+            var file = new File(path);
+
+            Assert.AreEqual(path, file.FullName);
+            Assert.AreEqual(parent, file.Parent.FullName);
+        }
+
+        [Test]
+        public void Init_PathIsNull()
+        {
+            Assert.That(() => new File(null), Throws.ArgumentNullException);
+        }
+
+        [Test]
+        public void Init_InvalidPath_InvalidDirectory()
+        {
+            if (SIO.Path.GetInvalidPathChars().Length == 0)
+            {
+                Assert.Ignore("This test does not make sense on systems where System.IO.Path.GetInvalidPathChars() returns an empty array.");
+            }
+
+            var path = SIO.Path.GetInvalidPathChars()[SIO.Path.GetInvalidPathChars().Length-1] + this.GetTestFileLocation();
+
+            Assert.That(() => new File(path), Throws.ArgumentException);
+        }
+
+        [Test]
+        public void Init_InvalidPath_InvalidFileName()
+        {
+            char invalidCharThatIsNotInInvalidPathChars = SIO.Path.GetInvalidFileNameChars().Except(SIO.Path.GetInvalidPathChars()).FirstOrDefault();
```

Comment:
I'm afraid there is no guarantee.
I changed the code so this test will be ignored if `invalidCharThatIsNotInInvalidPathChars` contains the default because the test will not make sense.